tracing-subscriber/tests/vec_subscriber_filter_interests_cached.rs

```diff
@@ -0,0 +1,117 @@
+#![cfg(feature = "registry")]
+mod support;
+use self::support::*;
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+use tracing::{Collect, Level};
+use tracing_subscriber::{filter, prelude::*};
+
+#[test]
+fn vec_subscriber_filter_interests_are_cached() {
+    let mk_filtered = |level: Level, subscriber: ExpectSubscriber| {
+        let seen = Arc::new(Mutex::new(HashMap::new()));
+        let filter = filter::filter_fn({
+            let seen = seen.clone();
+            move |meta| {
+                *seen.lock().unwrap().entry(*meta.level()).or_insert(0usize) += 1;
```

Comment:
it's up to you, but with nll, i think you can write this without all the combinators: just some old-fashioned `match` statements

Reply:
this code was copied from an existing test: https://github.com/tokio-rs/tracing/blob/baaab24b0d150136e9f4fe9bd1d7cfac62856ed9/tracing-subscriber/tests/multiple_subscriber_filter_interests_cached.rs#L13-L25

it _could_ also match on the `Entry`, like this:
```suggestion
                match seen.lock().unwrap().entry(*meta.level()) {
                    Entry::Occupied(occ) => occ.get_mut() += 1,
                    Entry::Vacant(vac) => vac.insert(1),
                }
```
but i thought the code with `or_insert` was actually _more_ readable, especially because we don't have to import the `hash_map::Entry` type. i could be convinced otherwise, though.